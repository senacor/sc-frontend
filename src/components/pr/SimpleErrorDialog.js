import React from 'react';
import { injectIntl } from 'react-intl';
import Dialog from '@material-ui/core/Dialog/Dialog';
import DialogTitle from '@material-ui/core/DialogTitle/DialogTitle';
import Typography from '@material-ui/core/Typography';
import { withStyles } from '@material-ui/core';

import PrStatusActionButton from './prDetail/PrStatusActionButton';

let styles = theme => ({
  inputClass: {
    position: 'relative',
    backgroundColor: theme.palette.primary['400'],
    color: theme.palette.secondary.white,
    marginBottom: '10%',
    marginTop: '5%',
    marginRight: '40%',
    marginLeft: '40%'
  },
  textClass: {
    textAlign: 'center',
    marginTop: '10%'
  }
});

const SimpleErrorDialog = ({
  classes,
  onClose,
  open,
  message,
  intl,
  ...other
}) => {
  return (
    <Dialog
      onClose={onClose}
      open={open}
      fullWidth={true}
      aria-labelledby="simple-dialog-title"
      {...other}
    >
      <DialogTitle>
        <Typography variant={'body2'} className={classes.textClass}>
          {message}
        </Typography>
      </DialogTitle>
      <PrStatusActionButton
        releaseButtonClick={onClose}
        label={intl.formatMessage({
          id: 'simpleerrordialog.ok'
        })}
        inputClass={classes.inputClass}
      />
    </Dialog>
  );
};

export default injectIntl(withStyles(styles)(SimpleErrorDialog));
