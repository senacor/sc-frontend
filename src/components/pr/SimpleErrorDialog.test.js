import { createShallow } from '@material-ui/core/test-utils';
import React from 'react';
import { SimpleErrorDialog } from './SimpleErrorDialog';

describe('SimpleErrorDialog Component', () => {
  let shallow = createShallow({ dive: true });

  it('should match Snapshot', () => {
    let component = shallow(
      <SimpleErrorDialog
        onClose={jest.fn()}
        open={true}
        message={'Fehlermeldung'}
      />
    );

    expect(component).toMatchSnapshot();
  });

  it('should call onClose if close', () => {
    let closeMock = jest.fn();
    let component = shallow(
      <SimpleErrorDialog
        onClose={closeMock}
        open={true}
        message={'Fehlermeldung'}
      />
    );

    component.find('WithStyles(Button)').simulate('click');
    expect(closeMock).toHaveBeenCalledTimes(1);
  });
});
