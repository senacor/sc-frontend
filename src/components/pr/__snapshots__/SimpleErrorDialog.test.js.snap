// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SimpleErrorDialog Component should match Snapshot 1`] = `
<Dialog
  TransitionComponent={[Function]}
  aria-labelledby="simple-dialog-title"
  classes={
    Object {
      "container": "MuiDialog-container-4",
      "paper": "MuiDialog-paper-5",
      "paperFullScreen": "MuiDialog-paperFullScreen-13",
      "paperFullWidth": "MuiDialog-paperFullWidth-12",
      "paperScrollBody": "MuiDialog-paperScrollBody-7",
      "paperScrollPaper": "MuiDialog-paperScrollPaper-6",
      "paperWidthLg": "MuiDialog-paperWidthLg-11",
      "paperWidthMd": "MuiDialog-paperWidthMd-10",
      "paperWidthSm": "MuiDialog-paperWidthSm-9",
      "paperWidthXs": "MuiDialog-paperWidthXs-8",
      "root": "MuiDialog-root-1",
      "scrollBody": "MuiDialog-scrollBody-3",
      "scrollPaper": "MuiDialog-scrollPaper-2",
    }
  }
  disableBackdropClick={false}
  disableEscapeKeyDown={false}
  fullScreen={false}
  fullWidth={false}
  maxWidth="sm"
  onClose={[Function]}
  open={true}
  scroll="paper"
  transitionDuration={
    Object {
      "enter": 225,
      "exit": 195,
    }
  }
>
  <WithStyles(DialogTitle)>
    Fehlermeldung
  </WithStyles(DialogTitle)>
  <WithStyles(Button)
    onClick={[Function]}
  >
    Ok
  </WithStyles(Button)>
</Dialog>
`;
